File: src/main/java/com/naminhyeok/fantazzk/template/domain/TemplateRoster.java
package com.naminhyeok.fantazzk.template.domain;

import java.util.List;
import java.util.Objects;

public final class TemplateRoster {
    private final List<String> names;

    private TemplateRoster(List<String> names) {
        this.names = List.copyOf(names);
    }

    public List<String> playerNames() {
        return names;
    }

    public static TemplateRoster exactlyRequired(List<String> playerNames, int requiredPlayerCount) {
        Objects.requireNonNull(playerNames, "playerNames must not be null");
        if (playerNames.size() != requiredPlayerCount) {
            throw new IllegalArgumentException("선수 수는 정확히 %d명이어야 합니다".formatted(requiredPlayerCount));
        }
        return new TemplateRoster(List.copyOf(playerNames));
    }
}

File: src/main/kotlin/com/naminhyeok/fantazzk/template/TemplateRoster.kt
package com.naminhyeok.fantazzk.template

class TemplateRoster private constructor(
    private val names: List<String>,
) {
    fun playerNames(): List<String> = names.toList()

    companion object {
        fun exactlyRequired(
            playerNames: List<String>,
            requiredPlayerCount: Int,
        ): TemplateRoster {
            require(playerNames.size == requiredPlayerCount) { "선수 수는 정확히 ${requiredPlayerCount}명이어야 합니다" }
            return TemplateRoster(playerNames.toList())
        }
    }
}
